File: server/Content/Flashcard.js
class Flashcard {
  constructor(front, back, topic = null) {
    this._front = front;
    this._back = back;
    this._topic = topic;
    this._showFront = true;
  }

  toString() {
    return `${this._front}, ${this._back}, ${this._topic}`;
  }

  get front() {
    return this._front;
  }

  set front(front) {
    this._front = front;
  }

  get back() {
    return this._back;
  }

  set back(back) {
    this._back = back;
  }

  get topic() {
    return this._topic;
  }

  set topic(topic) {
    this._topic = topic;
  }

  get showFront() {
    return this._showFront;
  }

  set showFront(showFront) {
    this._showFront = showFront;
  }

  flip() {
    this._showFront = !this._showFront;
    return this._showFront ? this._front : this._back;
  }
}

File: server/Content/Flashcard.py
class Flashcard:
    def __init__(self, front, back, topic=None):
        self.front = front
        self.back = back
        self.topic = topic
        self.showFront = True

    def __str__(self):
        return self.front(), self.back(), self.topic()
    
    @property
    def front(self):
        return self.front
    
    @front.setter
    def front(self, front):
        self.front = front
    
    @property
    def back(self):
        return self.back
    
    @back.setter
    def back(self, back):
        self.back = back
    
    @property
    def topic(self):
        return self.topic
    
    @topic.setter
    def topic(self, topic):
        self.topic = topic
    
    @property
    def showFront(self):
        return self.showFront
    
    @showFront.setter
    def showFront(self, showFront):
        self.showFront = showFront

    def flip(self):
        if (self.showFront()):
            self.showFront(False)
            return self.back()
        else:
            self.showFront(True)
            return self.front()
    
    
